### ISSUE TITLE
Bug - Can't enter intervention before date used for hospitalization

### ISSUE THREAD

**Issue by randalljam** [lemdt/CovidShinyModel#58]
Def think we need to fix this.

**Comment by jpspeng** [comment]
This isn't a bug - it's a logic problem with how we do our models. 

Say that we run the model and the hospitalizations on the date inputted aligns with Day 50 in the model. Then say that there was an intervention done 10 days ago. We set the intervention Re to go down on Day 40. When we run our models again then the date inputted no longer aligns with Day 50. So it can't work, at least the way that we're currently implementing.

**Comment by jpspeng** [comment]
Before, with Model 1, we could do this before since time from infection to hospitalization was preset and deterministic.  

For example, in the old model. Say time between infection and hospitalization is 10 days. Say that we run the model and the hospitalizations on the date inputted aligns with Day 50 in the model. Then say that there was an intervention done 6 days ago. This is completely fine because the change in Re from 6 days ago won't affect the hospitalization until 10 days later. The alignment wouldn't change.

**Comment by randalljam** [comment]
Good catch James, this is more complex than I thought, but I don't think it's a logical inconsistency. I think to fix we need to just iterate the alignment. I'm thinking through this in my excel model. This is a core functionality though - so I don't think we should punt this to phase2. I think it's the top priority this morning to fix for this Phase1 release.

**Comment by jpspeng** [comment]
I'm sorry but I'm going to push back on this. I think this requires some more thought. 

If we specify x number of people are in the hospital on some date, then we can't specify both a prior Re and an intervention Re in the past. There will be discrepancies in those four user inputs. 

I think we should either push back the Phase 1 release until Monday and spend some time thinking this through, or we should wait on this as a Phase 2 issue. 

Last minute changes without careful thought result in the inefficient changing of code back and forth, logic gaps start popping up, and bugs are bound to happen.
